id: PYSEC-2020-51
details: In jupyterhub-kubespawner before 0.12, certain usernames will be able to
  craft particular server names which will grant them access to the default server
  of other users who have matching usernames. This has been fixed in 0.12.
affected:
- package:
    name: jupyterhub-kubespawner
    ecosystem: PyPI
    purl: pkg:pypi:jupyterhub-kubespawner
  ranges:
  - type: GIT
    repo: https://github.com/jupyterhub/kubespawner
    events:
    - introduced: "0"
    - fixed: 3dfe870a7f5e98e2e398b01996ca6b8eff4bb1d0
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 0.12.0
  versions:
  - "0.1"
  - "0.5"
  - 0.5.1
  - 0.6.0
  - 0.7.1
  - "0.8"
  - 0.8.1
  - 0.9.0b1
  - 0.9.0b2
  - 0.9.0
  - 0.10.0
  - 0.10.1
  - 0.11.0
  - 0.11.1
references:
- type: ADVISORY
  url: https://github.com/jupyterhub/kubespawner/security/advisories/GHSA-v7m9-9497-p9gr
- type: FIX
  url: https://github.com/jupyterhub/kubespawner/commit/3dfe870a7f5e98e2e398b01996ca6b8eff4bb1d0
aliases:
- CVE-2020-15110
- GHSA-v7m9-9497-p9gr
modified: "2020-07-22T20:28:00Z"
published: "2020-07-17T21:15:00Z"
